Utilities/AtomicCounter.swift

```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+class AtomicCounter {
+  private let lock = NSLock()
+  private var _value = 0
```

Comment:
I see why you used the underscore in `Atomic<T>` for `_value` to disambiguate between the private property and the public one, but here I think the underscore is unnecessary since there's nothing to disambiguate and the property is entirely private.

Reply:
Sure